tool/resources/org/antlr/v4/tool/templates/codegen/Cpp/Cpp.stg

```diff
@@ -1048,8 +1048,8 @@ virtual antlrcpp::Any accept(tree::ParseTreeVisitor *visitor) override;
 VisitorDispatchMethod(method) ::=  <<
 
 antlrcpp::Any <parser.name>::<struct.name>::accept(tree::ParseTreeVisitor *visitor) {
-  if (dynamic_cast\<<parser.grammarName>Visitor*>(visitor) != nullptr)
-    return ((<parser.grammarName>Visitor *)visitor)->visit<struct.derivedFromName; format="cap">(this);
+  if (auto parserVisitor = dynamic_cast\<<parser.grammarName>Visitor*>(visitor))
```

Comment:
💭 I typically prefer to see the assignment on the line before the `if` (just so an assignment isn't in the conditional). That said:

1. This is generated code.
2. I'm not familiar enough with this target to know what it's coding conventions are. The form seen here may very well be preferred for this project.